facebook/react-native
# Fixes `react-native link` for Windows

Improve resilience of `link` command on Windows

There were various bugs related to `react-native link` for `react-native-windows`:    - NPM projects with multiple .csproj files could link the wrong projects
- Recent addition of ReactNativeHost (in lieu of ReactPage) was not supported
- Project GUIDs were not correctly referenced when linking dependencies
- NPM dependency projects needed to unnecessarily publish a Visual Studio solution file (i.e., .sln)
- Linking a project in a solution did not set up a default build configuration

This change set fixes all of the above. There are some remaining things to do for fit and finish:
- Support linking projects as well as their source code project (i.e., .csproj) dependencies to the RNW app solution

Fixes Microsoft/react-native-windows#738
Fixes Microsoft/react-native-windows#1576
Fixes Microsoft/react-native-windows#1578



## Motivation

The `react-native link` command is broken for version 0.50+ of `react-native-windows`.

## Test Plan

* Start localhost NPM registry (i.e., sinopia)
* Apply patch to `facebook/react-native` repo
* Publish patch to localhost registry
* Create a new project with `react-native init` from patched version of `react-native`
* Install a package with windows support (e.g., `react-native-fs`)
* Run `react-native link`
* Confirm link and unlink commands are successful for Windows.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/react-native-website, and link to your PR here.)

## Release Notes


[WINDOWS][BUGFIX][CLI] - Fix the link and unlink commands for `react-native-windows`
